### ISSUE TITLE
Error connection

### ISSUE THREAD

**Issue by GiuseppeChillemi** [gurzgri/red#1]
I have tried to connect to an ODBC data source called RD32, based on "SQL Server" driver. 

```
c: open ODBC://RD32
```

I receive the error: `.... access failed for user '' ` then the consolle closes

EDIT: Here is the output from RED `console.exe`
```
>> c: open odbc://rd32
28000 18456 [Microsoft][ODBC SQL Server Driver][SQL Server]Accesso non riuscito
per l'utente ''.
*** Script Error: ODBC error: ["28000" 18456 {[Microsoft][ODBC SQL Server Driver
][SQL Server]Accesso non riusc
*** Where: ctx||440~open-connection
*** Near : entity/spec/host
*** Stack:
```

Note that the USER DNS Origin has configured user and password inside

Also, I have tried the following:
```
connection: open make port! [
    scheme: 'odbc
    target: "driver={SQL Server};server=192.168.x.x;port=1433;database=ADB_RD32;uid=myuser;pwd=mypass"
]
```
(Server IP has been blanked)

The console hangs and I have to force close it.

SQL Server 2014 on Windows Server 2012 here

NOTE: One of the message I receive is "Invalid String Attribute" from the ODBC Driver. Removing the `port=1433` has solved the problem

**Comment by GiuseppeChillemi** [comment]
UPDATE:

I have had to:
change the server address to the textual one.
Remove the port configuration

Also I have moved from RED gui-console to console

This is the error I have:

>> do %test.red
01000 5701 [Microsoft][ODBC SQL Server Driver][SQL Server]Database context has been changed to 'ADB_RD32'.
01000 5703 [Microsoft][ODBC SQL Server Driver][SQL Server]Language setting has been changed to italian
*** Access Error: invalid UTF-8 encoding: #{C0C1C2C3}
*** Where: pick-information
*** Near : info
*** Stack: do-file

(Error strings translated by me)

**Comment by GiuseppeChillemi** [comment]
```
 probe system/schemes/odbc/state/drivers
 
[
    "Microsoft Access-Treiber (*.mdb)" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "2"
        "FileExtns" "*.mdb"
    ) 
    "Driver do Microsoft Paradox (*.db )" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "1"
        "FileExtns" "*.db"
    ) 
    "Driver do Microsoft Excel(*.xls)" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "1"
        "FileExtns" "*.xls"
    ) 
    "Microsoft Text Driver (*.txt; *.csv)" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "1"
        "FileExtns" "*.,*.asc,*.csv,*.tab,*.txt,*.csv"
    ) 
    {Driver da Microsoft para arquivos texto (*.txt; *.csv)} #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "1"
        "FileExtns" "*.,*.asc,*.csv,*.tab,*.txt,*.csv"
    ) 
    "Microsoft dBase-Treiber (*.dbf)" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "1"
        "FileExtns" "*.dbf,*.ndx,*.mdx"
    ) 
    "SQL Server" #(
        "CPTimeout" "60"
        "SQLLevel" "1"
        "DriverODBCVer" "03.50"
        "APILevel" "2"
        "ConnectFunctions" "YYY"
        "UsageCount" "1"
        "FileUsage" "0"
    ) 
    "Microsoft Excel Driver (*.xls)" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "1"
        "FileExtns" "*.xls"
    ) 
    "Driver do Microsoft dBase (*.dbf)" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "1"
        "FileExtns" "*.dbf,*.ndx,*.mdx"
    ) 
    "Microsoft Paradox-Treiber (*.db )" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "1"
        "FileExtns" "*.db"
    ) 
    "Microsoft ODBC for Oracle" #(
        "CPTimeout" "120"
        "SQLLevel" "1"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYY"
        "UsageCount" "1"
        "FileUsage" "0"
    ) 
    "Microsoft Text-Treiber (*.txt; *.csv)" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "1"
        "FileExtns" "*.,*.asc,*.csv,*.tab,*.txt,*.csv"
    ) 
    "Microsoft Excel-Treiber (*.xls)" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "1"
        "FileExtns" "*.xls"
    ) 
    "Microsoft Access Driver (*.mdb)" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "2"
        "FileExtns" "*.mdb"
    ) 
    "Driver do Microsoft Access (*.mdb)" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "2"
        "FileExtns" "*.mdb"
    ) 
    "Microsoft Paradox Driver (*.db )" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "1"
        "FileExtns" "*.db"
    ) 
    "Microsoft dBase Driver (*.dbf)" #(
        "SQLLevel" "0"
        "DriverODBCVer" "02.50"
        "APILevel" "1"
        "ConnectFunctions" "YYN"
        "UsageCount" "1"
        "FileUsage" "1"
        "FileExtns" "*.dbf,*.ndx,*.mdx"
    ) 
    "SQL Server Native Client 11.0" #(
        "UsageCount" "1"
        "APILevel" "2"
        "ConnectFunctions" "YYY"
        "CPTimeout" "60"
        "DriverODBCVer" "03.80"
        "FileUsage" "0"
        "SQLLevel" "1"
    ) 
    "ODBC Driver 11 for SQL Server" #(
        "UsageCount" "1"
        "APILevel" "2"
        "ConnectFunctions" "YYY"
        "CPTimeout" "60"
        "DriverODBCVer" "03.80"
        "FileUsage" "0"
        "SQLLevel" "1"
    ) 
    "Microsoft Visual FoxPro Driver" #(
        "UsageCount" "1"
        "FileExtns" "*.dbf,*.cdx,*.idx,*.fpt"
        "DriverODBCVer" "02.50"
        "FileUsage" "1"
        "APILevel" "0"
        "SQLLevel" "0"
        "ConnectFunctions" "YYN"
    ) 
    "Microsoft Visual FoxPro-Treiber" #(
        "UsageCount" "1"
        "FileExtns" "*.dbf,*.cdx,*.idx,*.fpt"
        "DriverODBCVer" "02.50"
        "FileUsage" "1"
        "APILevel" "0"
        "SQLLevel" "0"
        "ConnectFunctions" "YYN"
    ) 
    "Driver para o Microsoft Visual FoxPro" #(
        "UsageCount" "1"
        "FileExtns" "*.dbf,*.cdx,*.idx,*.fpt"
        "DriverODBCVer" "02.50"
        "FileUsage" "1"
        "APILevel" "0"
        "SQLLevel" "0"
        "ConnectFunctions" "YYN"
    ) 
    "Microsoft FoxPro VFP Driver (*.dbf)" #(
        "UsageCount" "1"
        "FileExtns" "*.dbf,*.cdx,*.idx,*.fpt"
        "DriverODBCVer" "02.50"
        "FileUsage" "1"
        "APILevel" "0"
        "SQLLevel" "0"
        "ConnectFunctions" "YYN"
    )
]
```

**Comment by GiuseppeChillemi** [comment]
```
>> probe system/schemes/odbc/state/sources
[
    "Excel Files" {Microsoft Excel Driver (*.xls, *.xlsx, *.xlsm, *.xlsb)} 
    "MS Access Database" "Microsoft Access Driver (*.mdb, *.accdb)" 
    "ArcaBUO-OK" "SQL Server" 
    "RD32" "SQL Server" 
    "RD32B" "SQL Server" 
    "Visual FoxPro Database" "Microsoft Visual FoxPro Driver" 
    "Visual FoxPro Tables" "Microsoft Visual FoxPro Driver"
]

**Comment by GiuseppeChillemi** [comment]
ODBC Panel

https://drive.google.com/file/d/1chhvrW7GTGDAJ6fmDK_FozfZ4D2t-6Bj/view?usp=sharing

Connection using IP Addess

connection: open make port! [
    scheme: 'odbc
    target: "driver={SQL Server};server=192.168.0.99;port=1433;database=ADB_RD32;uid=myuser;pwd=mypass"
]


Connection string using server name instead of IP

connection: open make port! [
    scheme: 'odbc
    target: "driver={SQL Server};server=SERVER\DB2014;port=1433;database=ADB_RD32;uid=myuser;pwd=mypass"
]

I Repeat the note written above:

One of the message I receive is "Invalid String Attribute" from the ODBC Driver. Removing the port=1433 has solved the problem

**Comment by gurzgri** [comment]
solved via commit fb1de49
